virttest/utils_net.py

```diff
@@ -2983,51 +2984,45 @@ def generate_mac_address_simple():
     return mac
 
 
-def get_ip_address_by_interface(ifname):
+def get_ip_address_by_interface(ifname, ip_ver="ipv4"):
     """
     returns ip address by interface
-    :param ifname - interface name
-    :raise NetError - When failed to fetch IP address (ioctl raised IOError.).
-
-    Retrieves interface address from socket fd trough ioctl call
-    and transforms it into string from 32-bit packed binary
-    by using socket.inet_ntoa().
-
+    :param ifname: interface name
+    :param ip_ver: Host IP version, ipv4 or ipv6.
+    :raise NetError: When failed to fetch IP address (ioctl raised IOError.).
     """
-    mysocket = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
     try:
-        return socket.inet_ntoa(fcntl.ioctl(
-            mysocket.fileno(),
-            arch.SIOCGIFADDR,
-            # ifname to binary IFNAMSIZ == 16
-            struct.pack('256s', ifname[:15])
-        )[20:24])
-    except IOError:
-        raise NetError(
-            "Error while retrieving IP address from interface %s." % ifname)
+        if ip_ver == "ipv6":
+            ver = netifaces.AF_INET6
+        else:
+            ver = netifaces.AF_INET
+        addr = netifaces.ifaddresses(ifname).get(ver)
+        return [a['addr'] for a in addr if any(s not in a['addr']
```

Comment:
if you want to get no linklocal address, you can do it like:
return [ a["addr"] for a in addr if not a['addr'].startswith('fe80')][0]